GateAdventures/Scenes/HelloScene.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using GateAdventures.Engine.Scenes;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using Microsoft.Xna.Framework.Input;
+
+namespace GateAdventures;
+
+// Example scene
+public class HelloScene : Scene
+{
+	private Vector2 _ballVelocity;
+	private Vector2 _ballPosition;
+	private Texture2D _ballTexture;
+	private MouseState _previousMouseState;
+
+	public override void Initialize(GameServiceContainer services)
+	{
+		base.Initialize(services);
+
+		_ballVelocity = Vector2.Zero;
+		_ballPosition = GraphicsDevice.Viewport.Bounds.Center.ToVector2();
+		_previousMouseState = Mouse.GetState();
+	}
+
+	public override void LoadContent()
+	{
+		_ballTexture = Content.Load<Texture2D>("RedBall");
+	}
+
+	public override void HandleInput()
+	{
+		MouseState currentMouseState = Mouse.GetState();
+		Vector2 mousePosition = new(currentMouseState.X, currentMouseState.Y);
+
+		// Calculate steering force with maximum magnitude of 4 in 1 frame
+		Vector2 desiredDirection = Vector2.Subtract(mousePosition, _ballPosition);
+		float desiredMagnitude = Math.Min(Vector2.Distance(mousePosition, _ballPosition), 4);
+		desiredDirection.Normalize();
+		desiredDirection = Vector2.Multiply(desiredDirection, desiredMagnitude);
+
+		// Apply steering force
+		_ballVelocity += desiredDirection;
+
+		// Limit velocity to a maximum magnitude of 25
+		float velocityMagnitude = Math.Min(Vector2.Distance(Vector2.Zero, _ballVelocity), 25);
+		_ballVelocity.Normalize();
+		_ballVelocity = Vector2.Multiply(_ballVelocity, velocityMagnitude);
+
+		// Apply friction
+		_ballVelocity = Vector2.Multiply(_ballVelocity, 0.9f);
+
+		if (_previousMouseState.LeftButton == ButtonState.Pressed && currentMouseState.LeftButton == ButtonState.Released)
+		{
+			SceneManager.Stop(this);
+			SceneManager.StartOverlay(new WorldScene());
+			return;
+		}
+
+		_previousMouseState = currentMouseState;
+	}
+
+	public override void Update(GameTime gameTime)
+	{
+		_ballPosition = Vector2.Add(_ballPosition, _ballVelocity);
+
+		// Horizontal bounds.
+		Rectangle bounds = GraphicsDevice.Viewport.Bounds;
+		if (_ballPosition.X < 0)
+		{
+			_ballPosition.X = 0;
+		}
+		else if (_ballPosition.X + 16 > bounds.Width)
+		{
+			_ballPosition.X = bounds.Width;
+		}
+
+		// Vertical bounds.
+		if (_ballPosition.Y < 0)
+		{
+			_ballPosition.Y = 0;
+		}
+		else if (_ballPosition.Y + 16 > bounds.Height)
+		{
+			_ballPosition.Y = bounds.Height;
+		}
+	}
+
+	public override void Draw(GameTime gameTime)
+	{
+		SpriteBatch.Begin();
+
+		SpriteBatch.Draw(_ballTexture, Vector2.Subtract(_ballPosition, new Vector2(16, 16)), new Rectangle(0, 0, 32, 32), Color.White);
+
+		SpriteBatch.End();
```

Comment:
I'm a bit nervous about each scene having it's own SpriteBatch. Seems like it defeats a lot of the point for why you would use a SpriteBatch in the first place, but I think it's probably fine for now. Just a point that we might want to discuss

Reply:
I will note(I need to verify though) if we want multiple shaders in the future you then need multiple sprite batch groups. This however is a shared instance between all scenes

Reply:
We could add a second SpriteBatch which you are expected to not call `Begin` or `End` with and can be used for anything that does not require a shader.